core/frontend/src/utils/pattern_validators.ts

```diff
@@ -0,0 +1,40 @@
+export function validate_port(port: number, public_range = true): boolean {
+  if (typeof port !== 'number') {
+    return false
+  }
+  if (!Number.isInteger(port)) {
+    return false
+  }
+  if (port > 65535) {
+    return false
+  }
+  if (public_range) {
+    return port >= 1024
+  }
+  return port >= 0
+}
+
+export function validate_integer_string(input: string): boolean {
+  return input.match(/^[0-9]+$/) != null
```

Comment:
What is the problem with `Number.isInteger()` ? Having a function only for that is not overkill ?
If you can't use this function with any type under typescript, It may be better to have `!isNaN(Number(input))` to be more explicit than using regex magic.

Reply:
`Number.isInteger()` will return `true` for both `111.0` and `111sasdad`. @Williangalvani noticed the problem and the solution I've found was to check for digit-only strings.

Also, typescript performs build-time tests only. It cannot validate runtime (e.g.: user-input) data.

Reply:
> Also, typescript performs build-time tests only. It cannot validate runtime (e.g.: user-input) data.

I was talking about `Number.isInteger` to receive an string, if that's a valid typescript input since has build-time analysis. 

And what about the second case ?

> If you can't use this function with any type under typescript, It may be better to have `!isNaN(Number(input))` to be more explicit than using regex magic.

Reply:
Just tested `!isNaN(Number(input))` and unfortunately it stills return `true` for the "1.0" case.